File: src/main/scala/com/dayet/Sqltest.scala
package com.dayet

import org.apache.spark.SparkConf
import org.apache.spark.sql.SparkSession
/**
  * @author Dayet
  * @date 2019/7/15 0:35
  * @version 1.0
  */
object Sqltest {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf()
    val spark = SparkSession.builder().config(conf).appName("aaa").master("local")
      .getOrCreate()
    val dataFrame = spark.read.format("json").load("")
    dataFrame.cache()
    val unit = dataFrame.createOrReplaceTempView("")



  }
}

File: src/main/java/sql/SQLtest.java
package sql;

import org.apache.spark.SparkConf;
import org.apache.spark.SparkContext;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SQLContext;

/**
 * @author Dayet
 * @version 1.0
 * @date 2019/7/24 21:54
 */
public class SQLtest {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf().setMaster("local").setAppName("sql");
        SparkContext sc = new SparkContext(conf);
        SQLContext scq = new SQLContext(sc);
        //不能读取嵌套的json
        Dataset<Row> dataset = scq.read().format("json").json("./json");
        dataset.registerTempTable("persion");
        dataset.createOrReplaceTempView("dada");

        Dataset<Row> sql = scq.sql("select name from persion");
        sql.show();
        dataset.show();

    }
}
